libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/service/MapboxTripServiceTest.kt

```diff
@@ -146,35 +141,42 @@ class MapboxTripServiceTest {
     }
 
     @Test
-    fun notificationDataPostToChannelOnServiceStarted() {
-        runBlocking {
-            service.startService()
-
-            val notificationData = MapboxTripService.getNotificationDataChannel().receive()
+    fun notificationDataObserverInvokedIfRegisteredBeforeServiceStart() {
+        val notificationDataObserver = mockk<NotificationDataObserver>(relaxUnitFun = true)
+        MapboxTripService.registerOneTimeNotificationDataObserver(notificationDataObserver)
+        service.startService()
 
-            assertEquals(NOTIFICATION_ID, notificationData.notificationId)
-            assertEquals(notification, notificationData.notification)
+        verify(exactly = 1) {
+            notificationDataObserver.onNotificationUpdated(
+                MapboxNotificationData(NOTIFICATION_ID, notification),
+            )
         }
     }
 
     @Test
-    fun channelClosedOnServiceStopped() {
-        runBlocking {
-            service.startService()
-            service.stopService()
-
-            // channel in MapboxTripService is private and static. Can't be mocked with Mockk.
-            // To check that the channel is closed after stopService we try to receive a new value
-            // and catch an exception.
-            try {
-                MapboxTripService.getNotificationDataChannel().receive()
-                fail()
-            } catch (e: Exception) {
-                assertTrue(e is CancellationException)
-            }
+    fun notificationDataObserverInvokedIfRegisteredAfterServiceStart() {
+        service.startService()
+        val notificationDataObserver = mockk<NotificationDataObserver>(relaxUnitFun = true)
+        MapboxTripService.registerOneTimeNotificationDataObserver(notificationDataObserver)
+
+        verify(exactly = 1) {
+            notificationDataObserver.onNotificationUpdated(
+                MapboxNotificationData(NOTIFICATION_ID, notification),
+            )
         }
     }
 
+    @Test
+    fun notificationDataObserverNotInvokedIfRegisteredAfterServiceStop() {
+        service.startService()
+        service.stopService()
+
+        val notificationDataObserver = mockk<NotificationDataObserver>(relaxUnitFun = true)
+        MapboxTripService.registerOneTimeNotificationDataObserver(notificationDataObserver)
+
+        verify(exactly = 0) { notificationDataObserver.onNotificationUpdated(any()) }
+    }
+
```

Comment:
Let's add a test that would invoke `notificationDataObserver.onNotificationUpdated(any())` multiple times thereby reflecting the new data.

Reply:
Actually `notificationDataObserver.onNotificationUpdated` is invoked only once, which the name `registerOneTimeNotificationDataObserver` suggests. We don't need to continuously observe notifications, we just need the first one. Otherwise `startForeground` will be called multiple times, which is not what we want.

Reply:
I updated the test to reflect that the observer is invoked not only once, but also with the latest data.